paddle/fluid/platform/mkldnn_reuse.h

```diff
@@ -644,14 +644,6 @@ class BinaryMKLDNNHandler : public platform::MKLDNNHandlerT<T, dnnl::binary> {
             platform::CreateKey(
                 dev_ctx, framework::vectorize(x->dims()), uniq_name,
                 (algo == dnnl::algorithm::binary_mul ? "M" : ""))) {
```

Comment:
Why aren't you just passing algo to CreateKey function? It can convert dnnl:algorithm datatype to a string